src/devices/TitanmecLed/TitanmecLedI2CLikeBase.cs

```diff
@@ -0,0 +1,261 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using System.Device;
+using System.Device.Gpio;
+using System.Runtime.InteropServices;
+
+namespace Iot.Device.TitanmecLed
+{
+    /// <summary>
+    /// Represents some Titanmec led devices which use I2C like communication protocol. This is an abstract class.
+    /// </summary>
+    /// <remarks>The communication protocol of some Titanmec led devices is like I2C, using one cable for data and one for clock, but not following the standard of I2C.</remarks>
+    public abstract class TitanmecLedI2CLikeBase :
+        ILedAdjustableBrightness, ILedCharactersOrder, 
```

Comment:
Interfaces specify the function which the device supported. The abstract class shows the class of the device, mostly for sharing code. Titanmec has dozens of chips, come with different form, communication protocol and functions. For example, there are several chips use serial protocol but support adjusting brightness in the same way, and some chips use 16-segment instead of the 7/8 one which use a pair of bytes for each segment (character) when displaying -- will not implement IDisplayOneByteCharacter. Currently, I don't want to bring more device into this project but I don't want to left trouble for user code sharing either.

At my side, an IoT device program is writing that need to support multiple kinds of LED solutions. Each kind of chip has its own initializing code block but I want to share the code in the main block using interface. I don't want to write multiple classes for changing brightness for every chips for example.

Reply:
Can we do minimal changes to add support for the new driver? I agree with @Ellerbach this is making it much more complicated - most of the interfaces should have a clear function and be obious how they should be implemented. Most of the interfaces introduced here they're neither obvious to implement nor they simplify code and they have no purpose for people using this code

Reply:
The simple proof of more complexity is the lines of code change in this PR: you've more than doubled the number of lines of code while providing one more binding which code is very similar to the previous one. I'd expect the change to add binding to be around 300 lines of code the most (+ maybe samples/docs)

Reply:
Interfaces are removed. Now all methods are declared in base class.